Add cli wrapper

trstruth opened the pull request:
so happy I learned about this repo today! I've wanted something like this to be able to practice against specific sequences I encounter in friendlies for a long time! I added a tiny cli wrapper, here's the usage:
```
Usage: tm_replay [OPTIONS] --slp-file <SLP_FILE> --start-frame <START_FRAME> --num-frames <NUM_FRAMES>

Options:
  -s, --slp-file <SLP_FILE>        Path to the input Slippi replay file
  -f, --start-frame <START_FRAME>  Start frame for the recording
  -n, --num-frames <NUM_FRAMES>    Number of frames to record
  -o, --output-file <OUTPUT_FILE>  Output filepath for the savestate [default: new_recording.gci]
  -n, --name <NAME>                Name to give to the recording (max 31 ASCII characters) [default: new_recording]
  -h, --help                       Print help
  -V, --version                    Print version
```

I tested this to work on a random fox/falco replay, and it imports into training mode no problem. Let me know what you think of this - happy to make whatever edits you deem necessary.

AlexanderHarrison commented:
- move the clap dep to dev-dependencies so that users depending on the lib don't pull it in.
- remove the unnecessary comments.
- the usage string when I give an invalid argument is "tm_replay <--slp-file <SLP_FILE>|--start-frame <START_FRAME>|--num-frames <NUM_FRAMES>|--output-file <OUTPUT_FILE>|--name <NAME>>" which is really weird. Should be the same as the help string
- use slp_parser::read_game instead of parse_file. parse_file does not work on slpz files.
- There's no need for the run function, just print the err and exit.

Thanks!

trstruth commented:
* `main.rs` requires its deps to be listed under `[dependencies]`, so what I've done instead is made a `[[bin]]` for the cli, which requires a feature, which requires the `clap` dependency, which is optional. I think this achieves the desired effect of clap being opt-in:
```
trstruth@Mars:~/projects/tm_replay$ cargo clean
     Removed 1319 files, 451.0MiB total
trstruth@Mars:~/projects/tm_replay$ cargo b
   Compiling libc v0.2.159
   Compiling shlex v1.3.0
   Compiling pkg-config v0.3.31
   Compiling memchr v2.7.4
   Compiling zstd-safe v7.2.1
   Compiling paste v1.0.15
   Compiling regex-syntax v0.8.5
   Compiling decancer v3.2.4
   Compiling lazy_static v1.5.0
   Compiling aho-corasick v1.1.3
   Compiling jobserver v0.1.32
   Compiling cc v1.1.24
   Compiling regex-automata v0.4.8
   Compiling zstd-sys v2.0.13+zstd.1.5.6
   Compiling regex v1.11.0
   Compiling zstd v0.13.2
   Compiling slpz v1.1.1
   Compiling slp_parser v0.1.0 (https://github.com/AlexanderHarrison/slp_parser.git#2397f319)
   Compiling tm_replay v0.1.0 (/home/trstruth/projects/tm_replay)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 3.02s
trstruth@Mars:~/projects/tm_replay$ cargo b --features cli
   Compiling proc-macro2 v1.0.86
   Compiling unicode-ident v1.0.13
   Compiling utf8parse v0.2.2
   Compiling is_terminal_polyfill v1.70.1
   Compiling colorchoice v1.0.2
   Compiling anstyle-query v1.1.1
   Compiling anstyle v1.0.8
   Compiling clap_lex v0.7.2
   Compiling strsim v0.11.1
   Compiling heck v0.5.0
   Compiling anstyle-parse v0.2.5
   Compiling anstream v0.6.15
   Compiling clap_builder v4.5.19
   Compiling quote v1.0.37
   Compiling syn v2.0.79
   Compiling clap_derive v4.5.18
   Compiling clap v4.5.19
   Compiling tm_replay v0.1.0 (/home/trstruth/projects/tm_replay)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 2.72s
   ```
   * most comments removed, I left in the docstrings above the cli args as they get populated in the `--help` output
   * removed the run function
   * updated to use `read_game` as opposed to `parse_file`
 
 regarding the message on invalid usage, I see the following (which is autogenne'd by clap):
![image](https://github.com/user-attachments/assets/f397b925-75f8-4f47-9524-87dd60c6d497)
this seems pretty reasonable to me, it prints an example valid usage and tells you to try `--help` for more info. Do you still want me to try to get it to print the --help page upon invalid usage? clap may expose some kind of option for that.

AlexanderHarrison commented:
Looks good. Just need to fix the weird argument error:

When I run: `cargo run --features=cli -- --slp-file ~/Slippi/Game_20240519T201317.slpz  --asd`

I get the error:
```
error: unexpected argument '--asd' found

Usage: tmreplay <--slp-file <SLP_FILE>|--start-frame <START_FRAME>|--num-frames <NUM_FRAMES>|--output-file <OUTPUT_FILE>|--name <NAME>>

For more information, try '--help'.
```

Can you replicate this?
